code/game/machinery/camera/camera_assembly.dm

```diff
@@ -132,19 +44,116 @@
 		user.drop_item()
 		I.loc = src
 		return
+	else
+		return ..()
 
-	// Taking out upgrades
-	else if(iscrowbar(I) && upgrades.len)
-		var/obj/U = locate(/obj) in upgrades
-		if(U)
-			to_chat(user, "<span class='notice'>You unattach an upgrade from the assembly.</span>")
-			playsound(loc, I.usesound, 50, 1)
-			U.loc = get_turf(src)
-			upgrades -= U
+/obj/item/camera_assembly/crowbar_act(mob/user, obj/item/I)
+	if(!upgrades.len)
+		return
+	. = TRUE
+	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+		return
+	var/obj/U = locate(/obj) in upgrades
+	if(U)
+		to_chat(user, "<span class='notice'>You unattach an upgrade from the assembly.</span>")
+		playsound(loc, I.usesound, 50, 1)
+		U.loc = get_turf(src)
+		upgrades -= U
+
+/obj/item/camera_assembly/screwdriver_act(mob/user, obj/item/I)
+	if(state != ASSEMBLY_WIRED)
+		return
+	. = TRUE
+	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+		return
+	state = HEY_IM_WORKING_HERE
+	var/input = strip_html(input(usr, "Which networks would you like to connect this camera to? Seperate networks with a comma. No Spaces!\nFor example: SS13,Security,Secret ", "Set Network", "SS13"))
+	if(!input)
+		state = ASSEMBLY_WIRED
+		to_chat(usr, "<span class='warning'>No input found please hang up and try your call again.</span>")
+		return
+
+	var/list/tempnetwork = splittext(input, ",")
+	if(tempnetwork.len < 1)
+		state = ASSEMBLY_WIRED
+		to_chat(usr, "<span class='warning'>No network found please hang up and try your call again.</span>")
+		return
+
+	var/area/camera_area = get_area(src)
+	var/temptag = "[sanitize(camera_area.name)] ([rand(1, 999)])"
+	input = strip_html(input(usr, "How would you like to name the camera?", "Set Camera Name", temptag))
+	state = ASSEMBLY_BUILT
+	var/obj/machinery/camera/C = new(loc)
+	loc = C
+	C.assembly = src
+
+	C.auto_turn()
+
+	C.network = uniquelist(tempnetwork)
+	tempnetwork = difflist(C.network,GLOB.restricted_camera_networks)
+	if(!tempnetwork.len) // Camera isn't on any open network - remove its chunk from AI visibility.
+		cameranet.removeCamera(C)
+
+	C.c_tag = input
+
+	for(var/i = 5; i >= 0; i -= 1)
+		var/direct = input(user, "Direction?", "Assembling Camera", null) in list("LEAVE IT", "NORTH", "EAST", "SOUTH", "WEST" )
+		if(direct != "LEAVE IT")
+			C.dir = text2dir(direct)
+		if(i != 0)
+			var/confirm = alert(user, "Is this what you want? Chances Remaining: [i]", "Confirmation", "Yes", "No")
+			if(confirm == "Yes")
+				break
+
+
+/obj/item/camera_assembly/wirecutter_act(mob/user, obj/item/I)

```

Comment:
You know, seeing so many of these similar procs during this refactor makes me wonder if we need a default construction/deconstruction proc, and if so if it should wait for another PR. Thoughts @Fox-McCloud?

Reply:
Explain a bit further?

Reply:
For example, in order to take down a reinforced wall currently there are a whole bunch of code blocks that are nearly identical (i.e. you apply a tool to the wall, it gives a couple of messages, takes some time, and if successful it changes a variable in the wall or deletes it). There are other similar construction steps in many other places e.g. mech construction, camera construction.

My thought is that it's probably quite easy to come up with a standard proc to use for all atoms that have construction/deconstruction steps, which should clean up the code a bit and cut down on future work

Reply:
maybe in the future, but probably not in this PR.

Construction can vary so much, as well, that it may be difficult to implement properly.

Reply:
Fair enough! I'll leave that for now then.